src/astro/files/locations/__init__.py

```diff
@@ -16,14 +16,38 @@
 def create_file_location(path: str, conn_id: Optional[str] = None) -> BaseFileLocation:
     """
     Location factory method to generate location class
-
     :param path: Path to a file in the filesystem/Object stores
     :param conn_id: Airflow connection ID
     """
     filetype: FileLocation = BaseFileLocation.get_location_type(path)
     module_path = DEFAULT_CONN_TYPE_TO_MODULE_PATH[filetype.value]
-    module_name = module_path.split(".")[-1]
-    class_name = module_name.title() + "Location"
     module_ref = importlib.import_module(module_path)
+    class_name = get_class_name(module_ref)
     location: BaseFileLocation = getattr(module_ref, class_name)(path, conn_id)
     return location
+
+
+def get_class_name(module_ref: Any, suffix: str = "Location") -> str:
```

Comment:
Perhaps this function could be placed in `astro.utils.path` or somewhere else, so we can reuse it in other parts of the code.

Reply:
Moved and updated the database package to use it.